src/content/docs/zh-tw/basics/astro-components.mdx

```diff
@@ -0,0 +1,394 @@
+
+
+
+### 具名插槽
+
+Astro 元件可以放置具名插槽，讓你只傳入名字相符的 HTML 元素到指定插槽。
+
+具名插槽使用 `name` 屬性：
+
+```astro /<slot .*?/>/
+---
+// src/components/Wrapper.astro
+import Header from './Header.astro';
+import Logo from './Logo.astro';
+import Footer from './Footer.astro';
+
+const { title } = Astro.props;
+---
+<div id="content-wrapper">
+  <Header />
+  <slot name="after-header" />  <!-- 有 `slot="after-header"` 屬性的子元素會跑來這 -->
+  <Logo />
+  <h1>{title}</h1>
+  <slot />  <!-- 沒有 `slot` 屬性的子元素，以及有 `slot="default"` 屬性的子元素會跑來這 -->
+  <Footer />
+  <slot name="after-footer" />  <!-- 有 `slot="after-footer"` 屬性的子元素會跑來這 -->
+</div>
+```
+
+要把 HTML 內容放進特定的插槽，必須在子元素加上 `slot` 屬性指定對應的插槽，所有其他子元素則會被放入預設（不具名）的 `<slot />`。
+
+```astro /slot=".*?"/
+---
+// src/pages/fred.astro
+import Wrapper from '../components/Wrapper.astro';
+---
+<Wrapper title="Fred's Page">
+  <img src="https://my.photo/fred.jpg" slot="after-header" />
+  <h2>All about Fred</h2>
+  <p>Here is some stuff about Fred.</p>
+  <p slot="after-footer">Copyright 2022</p>
+</Wrapper>
+```
+
+:::tip
+在子元素加上 `slot="my-slot"`，藉此將它們傳遞到對應的 `<slot name="my-slot" />` 預留位置。
+:::
+
+若要一次傳遞多個 HTML 元素，但不想額外包一層 `<div>` 的話，可以在 [Astro 的 `<Fragment/>` 元件](/zh-tw/reference/api-reference/#fragment-)使用 `slot=""`：
+
+```astro title="src/components/CustomTable.astro" "<slot name="header"/>" "<slot name="body"/>"
+---
+// 客製一個 table，使用具名插槽為 head 和 body 預留兩個位置
+---
+<table class="bg-white">
+  <thead class="sticky top-0 bg-white"><slot name="header" /></thead>
+  <tbody class="[&_tr:nth-child(odd)]:bg-gray-100"><slot name="body" /></tbody>
+</table>
+```
+
+透過 `slot=""` 屬性，將表格內容放置於 `"header"` 和 `"body"`。還能單獨調整 HTML 元素的樣式。
+
+```astro title="src/components/StockTable.astro" {5-7, 9-13} '<Fragment slot="header">' '<Fragment slot="body">'
+---
+import CustomTable from './CustomTable.astro';
+---
+<CustomTable>
+  <Fragment slot="header"> <!-- 表首會傳到這 -->
+    <tr><th>Product name</th><th>Stock units</th></tr>
+  </Fragment>
+
+  <Fragment slot="body"> <!-- 內容會傳到這 -->
+    <tr><td>Flip-flops</td><td>64</td></tr>
+    <tr><td>Boots</td><td>32</td></tr>
+    <tr><td>Sneakers</td><td class="text-red-500">0</td></tr>
+  </Fragment>
+</CustomTable>
+```
+
+只能透過具名插槽傳送一層子元素，不能傳送巢狀元素，請特別留意。
+
+:::tip
+具名插槽也可傳遞到 [UI 框架元件](/zh-tw/guides/framework-components/)！
+:::
+
+
+:::note
+Astro 插槽名稱無法動態產生（例如透過 map 函式操作）。想在 UI 框架元件裡動態產生插槽名稱的話，最好還是讓框架本身去處理。
+:::
+
+
+### 插槽的備用內容
+
+插槽也能渲染**備用內容**。沒有對應的子元素傳遞過來時，`<slot />` 元素會渲染自己的備用內容。
+
+```astro {14}
+---
+// src/components/Wrapper.astro
+import Header from './Header.astro';
+import Logo from './Logo.astro';
+import Footer from './Footer.astro';
+
+const { title } = Astro.props;
+---
+<div id="content-wrapper">
+  <Header />
+  <Logo />
+  <h1>{title}</h1>
+  <slot>
+    <p>This is my fallback content, if there is no child passed into slot</p>
+  </slot>
+  <Footer />
+</div>
+```
+
+### Transferring slots
```

Comment:
Forgot to translate here, or kept it on purpose?

Reply:
Oops, I lost this one (and another one). Translated here: https://github.com/withastro/docs/pull/6948/commits/095774d7a7d84725f3b3d24af75cc1fe1cf5d7eb